archinstall/lib/user_interaction.py

```diff
@@ -49,8 +49,15 @@ def print_large_list(options, padding=5, margin_bottom=0, separator=': '):
 
 def ask_for_superuser_account(prompt='Create a required super-user with sudo privileges: ', forced=False):
 	while 1:
-		new_user = input(prompt).strip(' ').lower()
-		
+		new_user = input(prompt).strip(' ')
+
+		if not re.match('[a-z_][a-z0-9_-]*[$]?', new_user) or len(new_user) > 32:
```

Comment:
I think there's a `^` needed at the start.

Reply:
You're right. Updated this and fixed an issue where it accepts a username with a last character that is not a dollar sign